### ISSUE TITLE
Sc-ide vs sclang handshaking doesn't complete if library didn't compile the first time

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#5864]
## Environment

* SuperCollider version: 3.13.0-dev but I'm sure it affects earlier versions
* Operating system: Linux (I assume doesn't matter)
* Other details (Qt version, audio driver, etc.): n/a

## Steps to reproduce

I just encountered a behavior that suggests a bug in the sclang <--> sc-ide handshaking.

1. Do something to the class library to introduce a compile error. (In my case, it was adding a directory to the include paths, but you could introduce an error manually by editing any class file.)
2. Reboot the interpreter.
3. Correct the class library error by editing (not by using the Preferences panel).
4. Recompile the class library.

Now the IDE will fail to connect -- for instance, you'll get an error from `Document.globalKeyDownAction = { ... something ... }`.

## Expected vs. actual behavior

This suggests that part of the editor-interpreter handshaking takes place once and only once, when the sclang process starts up, and that this depends on successful library compilation. I would guess that the IDE does not keep track of success or failure, and as a result, doesn't know to retry that part of the handshaking upon a successful library recompile. (Sure, it's a subtle point, and maybe not critical, but, if there is a possibility of part of the init process failing to complete, then formally it should be retried.)

If this is too difficult to fix, it might be helpful to document somewhere that the sequence "1/ reboot interpreter; 2/ fail; 3/ fix library; 4/ recompile library" is not safe and that step 4 in this case should be "reboot the interpreter again."

**Comment by loopier** [comment]
I stumbled with a problem that might be related to this, but I'm not sure when it started (maybe when I upgraded to `3.12.2-3`).

I was editing my own class that works only with MIDI, and at some point the communication between `sclang <-> scserver` broke. I can boot the server with `s.boot` and don't get any errors:
```
-> localhost
Booting server 'localhost' on address 127.0.0.1:57110.
Found 115 LADSPA plugins
Faust: supercollider.cpp: sc_api_version = 3
Faust: FaustJPverbRaw numControls=11
Faust: supercollider.cpp: sc_api_version = 3
Faust: FaustGreyholeRaw numControls=7
JackDriver: client name is 'SuperCollider'
SC_AudioDriver: sample rate = 48000.000000, driver's block size = 1024
JackDriver: connected  UMC404HD 192k Analog Surround 4.0:capture_FL to SuperCollider:in_1
JackDriver: connected  UMC404HD 192k Analog Surround 4.0:capture_FR to SuperCollider:in_2
JackDriver: connected  SuperCollider:out_1 to UMC404HD 192k Analog Surround 4.0:playback_FL
JackDriver: connected  SuperCollider:out_2 to UMC404HD 192k Analog Surround 4.0:playback_FR
SuperCollider 3 server ready.
JackDriver: max output latency 32.0 ms
```
But the server window doesn't show up it's booted (stays as when it fails -- yellow and all values set to 0).
When trying to play something like `{SinOsc.ar * 0.1}.play` I get a warning and no sound:
```
WARNING: server 'localhost' not running.
-> nil
```
I've killed lurking scsynth instances, tried rebooting, removing all plug-ins and quarks and the problem persists.

I'm on `Linux localhost 5.19.9-arch1-1 x86_64 GNU/Linux ` using Pipewire (haven't tried with Jack, but the server doesn't complain and it worked fine until a couple days ago). `scsynth` does actually start and is listed in running processes on the task manager.

I wasn't sure where to report this. I'm not sure if it's an Arch package issue (I installed it with `pacman` from the offical Arch repo) or an upstream bug. If this isn't the place, please let me know and I'll repost elsewhere (pointers welcome).

**Comment by jamshark70** [comment]
> I stumbled with a problem that might be related to this

With apologies, but it isn't related.

The present bug report is for scide (editor) vs language communication. The server is not involved in this bug report at all.

Your issue might be a SuperCollider bug, but it belongs in a new issue, not this one.
